contracts/StorageAccessible.sol

```diff
@@ -52,6 +52,30 @@ contract StorageAccessible {
         }
     }
 
+    /**
+     * @dev Same as simulateDelegatecall but with view modifier (only uses static context)
+     * @param targetContract Address of the contract containing the code to execute.
+     * @param calldataPayload Calldata that should be sent to the target contract (encoded method name and arguments).
+     */
+    function simulateStaticDelegatecall(
+        address targetContract,
+        bytes memory calldataPayload
+    ) public view returns (bytes memory) {
+        bytes memory innerCall = abi.encodeWithSelector(
```

Comment:
We duplicate quite some code here (only difference is that we use staticcall instead of call) maybe this can be optimized with a simple if.

Reply:
How exactly would you envision the code to be shared?

Most of the duplication comes from encoding the call and converting the response either in revert message or success bytes.

If we want to declare a `view` flavour of the simulate call method, we can only use staticCall inside (thus the favor that is non view version would have to be a separate method). As soon as we use a staticCall we have to again transform the success bit and response bytes so I'm not sure we can actually save much.